runtime/Cpp/runtime/src/atn/PredictionContext.cpp

```diff
@@ -233,15 +204,13 @@ Ref<PredictionContext> PredictionContext::mergeRoot(const Ref<SingletonPredictio
       return EMPTY;
     }
     if (a == EMPTY) { // $ + x = [$,x]
-      std::vector<size_t> payloads = { b->returnState, EMPTY_RETURN_STATE };
-      std::vector<Ref<PredictionContext>> parents = { b->parent, nullptr };
-      Ref<PredictionContext> joined = std::make_shared<ArrayPredictionContext>(parents, payloads);
+      std::vector<PredictionContextItem> contexts = {PredictionContextItem(b->parent, b->returnState), PredictionContextItem(nullptr, EMPTY_RETURN_STATE)};
+      Ref<PredictionContext> joined = std::make_shared<ArrayPredictionContext>(contexts);
       return joined;
     }
     if (b == EMPTY) { // x + $ = [$,x] ($ is always first if present)
```

Comment:
Can anybody explain the history of this comment? It appears to contradict the code (in that $ is always added second if present). Its also in the Java run-time. The contradiction should be removed.

Reply:
This is something Ter or Sam must answer.